addons/mail/static/src/discuss/call/common/call.js

```diff
@@ -72,16 +74,41 @@ export class Call extends Component {
         const invitationCards = [];
         const filterVideos = this.props.thread.showOnlyVideo && this.props.thread.videoCount > 0;
         for (const session of Object.values(this.props.thread.rtcSessions)) {
-            if (!filterVideos || session.videoStream) {
-                const data = {
-                    key: "session_" + session.id,
-                    session,
-                };
-                if (session.raisingHand) {
-                    raisingHandCards.push(data);
-                } else {
-                    sessionCards.push(data);
-                }
+            if (filterVideos && !session.videoStream.size) {
+                continue;
+            }
+            const data = {
+                key: "session_" + session.id,
+                session,
+            };
+            if (session.raisingHand) {
+                raisingHandCards.push(data);
+                continue;
+            }
+            if (!session.videoStreamByType.size) {
+                sessionCards.push(data);
+                continue;
+            }
+            switch (session.videoStreamByType.size) {
+                case 1:
+                    if (session.videoStreamByType.keys().next().value === "screen") {
+                        sessionCards.push(data);
+                    }
+                    sessionCards.push({
+                        key: "session_" + session.id + "screen",
+                        session,
+                        videoStream: session.videoStreamByType.values().next().value,
+                    });
+                    break;
+                case 2:
+                    for (const [videoType, stream] of session.videoStreamByType.entries()) {
+                        sessionCards.push({
+                            key: "session_" + session.id + videoType,
+                            session,
+                            videoStream: stream,
+                        });
+                    }
+                    break;
             }
```

Comment:
This is what is causing the bug that paju reported on the task:

> the "raise hand" feature, the camera is deactivated when using it (no issue in master)

You do a `continue` if it's a raising hand card, which then ignores the addition of video stream(s) here.


I wouldn't put a switch like that but instead format what is now called `data` before adding it to one of the card array:

```js
for (const session of Object.values(this.props.thread.rtcSessions)) {
  const newCards = [];
  // this formats all the cards of one session, regardless of the amount of streams that it currently has
  if (!session.videoStreamByType.size) {
    if (filterVideos) {
        continue;
    }
    newCards.push({
        key: "session_" + session.id,
        session,
    });
  } else {
    for (const [videoType, stream] of session.videoStreamByType.entries()) {
        newCards.push({
            key: "session_" + session.id + videoType,
            session,
            videoStream: stream,
        });
    }
  }
  if (session.raisingHand) {
      raisingHandCards.push(...newCards);
      continue;
  }
  if (!session.videoStreamByType.size) {
      sessionCards.push(...newCards);
      continue;
  }
}

Reply:
if you do the discord-like spec where the webcam is a substitute for the avatar card:

```js
for (const session of Object.values(this.props.thread.rtcSessions)) {
    const target = session.raisingHand ? raisingHandCards : sessionCards;

    // card 1
    target.push({
        key: "session_main_" + session.id,
        session,
        videoStream: session.videoStreamByType.get("camera"),
    });

    // card 2
    const screenStream = session.videoStreamByType.get("screen");
    if (screenStream) {
        target.push({
            key: "session_secondary_" + session.id,
            session,
            videoStream: screenStream,
        });
    }
}
```